src/renderer/components/+workloads-overview/overview.tsx

```diff
@@ -15,60 +14,23 @@ import { replicaSetStore } from "../+workloads-replicasets/replicasets.store";
 import { jobStore } from "../+workloads-jobs/job.store";
 import { cronJobStore } from "../+workloads-cronjobs/cronjob.store";
 import { Events } from "../+events";
-import { KubeObjectStore } from "../../kube-object.store";
 import { isAllowedResource } from "../../../common/rbac";
-import { namespaceStore } from "../+namespaces/namespace.store";
+import { kubeWatchApi } from "../../api/kube-watch-api";
 
 interface Props extends RouteComponentProps<IWorkloadsOverviewRouteParams> {
 }
 
 @observer
 export class WorkloadsOverview extends React.Component<Props> {
-  @observable isLoading = false;
-  @observable isUnmounting = false;
-
-  async componentDidMount() {
-    const stores: KubeObjectStore[] = [
-      isAllowedResource("pods") && podsStore,
-      isAllowedResource("deployments") && deploymentStore,
-      isAllowedResource("daemonsets") && daemonSetStore,
-      isAllowedResource("statefulsets") && statefulSetStore,
-      isAllowedResource("replicasets") && replicaSetStore,
-      isAllowedResource("jobs") && jobStore,
-      isAllowedResource("cronjobs") && cronJobStore,
-      isAllowedResource("events") && eventStore,
-    ].filter(Boolean);
-
-    const unsubscribeMap = new Map<KubeObjectStore, () => void>();
-
-    const loadStores = async () => {
-      this.isLoading = true;
-
-      for (const store of stores) {
-        if (this.isUnmounting) break;
-
-        try {
-          await store.loadAll();
-          unsubscribeMap.get(store)?.(); // unsubscribe previous watcher
-          unsubscribeMap.set(store, store.subscribe());
-        } catch (error) {
-          console.error("loading store error", error);
-        }
-      }
-      this.isLoading = false;
-    };
-
-    namespaceStore.onContextChange(loadStores, {
-      fireImmediately: true,
-    });
-
-    await when(() => this.isUnmounting && !this.isLoading);
-    unsubscribeMap.forEach(dispose => dispose());
-    unsubscribeMap.clear();
-  }
-
-  componentWillUnmount() {
-    this.isUnmounting = true;
+  componentDidMount() {
+    disposeOnUnmount(this, [
+      kubeWatchApi.subscribeStores([
```

Comment:
Why aren't we checking `isAllowedResource` anymore?

Reply:
Because now it's handled under the hood in `KubeObjectStore.loadItems()`.
Also notice that `KubeObjectStore.subscribe()` does filtering for allowed apis to subscribe.